File: examples/01-linear/python/main.py
# Example 01: linear order-processing pipeline
# ValidateOrder → CalculateTax → ChargePayment → SendConfirmation → end
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "../../../sdk/python"))
import kflow

from handlers import (
    STATE_VALIDATE_ORDER,
    STATE_CALCULATE_TAX,
    STATE_CHARGE_PAYMENT,
    STATE_SEND_CONFIRMATION,
    validate_order,
    calculate_tax,
    charge_payment,
    send_confirmation,
)


def main():
    wf = kflow.Workflow("order-processing")
    wf.task(STATE_VALIDATE_ORDER)(validate_order)
    wf.task(STATE_CALCULATE_TAX)(calculate_tax)
    wf.task(STATE_CHARGE_PAYMENT)(charge_payment)
    wf.task(STATE_SEND_CONFIRMATION)(send_confirmation)

    wf.flow(
        kflow.step(STATE_VALIDATE_ORDER).next(STATE_CALCULATE_TAX),
        kflow.step(STATE_CALCULATE_TAX).next(STATE_CHARGE_PAYMENT),
        kflow.step(STATE_CHARGE_PAYMENT).next(STATE_SEND_CONFIRMATION),
        kflow.step(STATE_SEND_CONFIRMATION).end(),
    )

    print("=== 01-linear: order-processing ===")
    kflow.run_local(wf, {"order_id": "ORD-2001", "customer": "alice", "total": 149.99})
    print("  → workflow COMPLETED")


if __name__ == "__main__":
    main()

File: examples/01-linear/main.go
// Example 01: linear order-processing pipeline
// ValidateOrder → CalculateTax → ChargePayment → SendConfirmation → end
package main

import (
	"fmt"
	"log"

	"github.com/pastorenue/kflow/examples/01-linear/handlers"
	"github.com/pastorenue/kflow/internal/local"
	"github.com/pastorenue/kflow/pkg/kflow"
)

func main() {
	h := handlers.New()

	wf := kflow.New("order-processing")
	wf.Task(handlers.StateValidateOrder, h.ValidateOrder)
	wf.Task(handlers.StateCalculateTax, h.CalculateTax)
	wf.Task(handlers.StateChargePayment, h.ChargePayment)
	wf.Task(handlers.StateSendConfirmation, h.SendConfirmation)

	wf.Flow(
		kflow.Step(handlers.StateValidateOrder).Next(handlers.StateCalculateTax),
		kflow.Step(handlers.StateCalculateTax).Next(handlers.StateChargePayment),
		kflow.Step(handlers.StateChargePayment).Next(handlers.StateSendConfirmation),
		kflow.Step(handlers.StateSendConfirmation).End(),
	)

	fmt.Println("=== 01-linear: order-processing ===")
	if err := local.RunLocal(wf, kflow.Input{
		"order_id": "ORD-2001",
		"customer": "alice",
		"total":    149.99,
	}); err != nil {
		log.Fatalf("workflow failed: %v", err)
	}
	fmt.Println("  → workflow COMPLETED")
}
